atmos/modules/helmfile/helmfile-core.variant

```diff
@@ -66,6 +66,23 @@ job "helmfile subcommand" {
     }
   }
 
+  config "config-component" {
+    source job {
+      name = "stack config"
+      args = {
+        component-type = "helmfile"
+        config-type    = "component"
+        component      = param.component
+        stack          = opt.stack
+        format         = "json"
+      }
+    }
+  }
+
+  variable "component" {
+    value = coalesce(conf.config-component.component, param.component)
```

Comment:
I am not completely following the Variant code. How is it that `conf.config-component.component` would be empty but `param.component` would not, give line 75 above?

Reply:
`param.component` is what is coming from the CLI request (ex: `nginx-ingress2`) so it looks in the stack config for that yaml component. It then looks for a custom component (ex: `component: nginx-ingress`) which is `conf.config-component.component`, but it could be missing so we fall back to `param.component`.

Reply:
for a regular component (w/o `component` attribute), `conf.config-component.component` will be empty, but `param.component` you provide then you run `atmos helmfile diff xxxx`.
This is the same as we have for terraform.